test/utils/setup.ts

```diff
@@ -1,14 +1,50 @@
 /*
     Setup -- start dynamodb instance
  */
+import { spawn } from 'child_process'
+const waitPort = require('wait-port')
 import DynamoDbLocal from 'dynamo-db-local'
 
 const PORT = 4567
 
 module.exports = async () => {
+    let dynamodb
 
-    let dynamodb = DynamoDbLocal.spawn({port: PORT})
-    process.env.DYNAMODB_PID = dynamodb.pid
-    process.env.DYNAMODB_PORT = PORT.toString()
-    // console.log('Spawn DynamoDB', dynamodb.pid)
+    if (String(process.env.DOCKER) === 'true') {
+        const args = [`run`, `-p`, `${PORT}:8000`, `amazon/dynamodb-local`]
+        console.info('\nUsing docker to run dynamoDB', {
+            args,
+        })
+
+        // Docker errors will be forwarded to the local terminal with
+        // stdio: inherit
+        dynamodb = spawn(`docker`, args, {
+            cwd: __dirname,
+            stdio: 'inherit',
+        })
+
+        await waitPort({
+            host: '0.0.0.0',
+            port: PORT,
+            timeout: 3000,
+        })
+
+        console.info('Docker is ready')
+    } else {
+        console.info('Using local Java to run dynamoDB')
+        dynamodb = DynamoDbLocal.spawn({ port: PORT })
+    }
+
+    process.env.DYNAMODB_PID = String(dynamodb.pid)
+    process.env.DYNAMODB_PORT = String(PORT)
+
+    // When jest throws anything unhandled, ensure we kill the spawned process
+    process.on('unhandledRejection', (error) => {
```

Comment:
This should ensure we don't leave the spawned process running in the event of anything nasty happening.

Reply:
Would it be better to pass in process.env.DOCKER the command line so the port would not be hard coded?

Reply:
Answered in https://github.com/sensedeep/dynamodb-onetable/pull/97/files#r690077693; let me  know, I can always make the change, I am  just trying to understand it :+1: